test/TestMath.sol

```diff
@@ -53,4 +54,54 @@ contract TestMath is Test {
         vm.assume(y > 0 && x % y != 0);
         assertEq(mock.divUp(x, y), x / y + 1);
     }
+
+    function testLog2CompareNaive(uint256 x) public {
+        assertEq(mock.log2(x), ref.log2Naive(x));
+    }
+
+    function testLog2CompareDicho(uint256 x) public {
+        assertEq(mock.log2(x), ref.log2Dichotomy(x));
+    }
+
+    function testDeBruijnSequence() public {
+        uint256 deBruijnSeq = 0x00818283848586878898a8b8c8d8e8f929395969799a9b9d9e9faaeb6bedeeff;
+        assertTrue(DeBruijn.isDeBruijnSequence(deBruijnSeq));
+    }
+
+    function testHashTable() public {
+        uint256 m;
+        uint256 key;
+        uint256 value;
+        uint256 deBruijnSeq = 0x00818283848586878898a8b8c8d8e8f929395969799a9b9d9e9faaeb6bedeeff;
+        assembly {
+            m := mload(0x40)
+            mstore(m, 0x0001020903110a19042112290b311a3905412245134d2a550c5d32651b6d3a75)
+            mstore(add(m, 0x20), 0x06264262237d468514804e8d2b95569d0d495ea533a966b11c886eb93bc176c9)
+            mstore(add(m, 0x40), 0x071727374353637324837e9b47af86c7155181ad4fd18ed32c9096db57d59ee3)
+            mstore(add(m, 0x60), 0x0e2e4a6a5f92a6be3498aae067ddb2eb1d5989b56fd7baf33ca0c2ee77e5caf7)
+            mstore(add(m, 0x80), 0xff0810182028303840444c545c646c7425617c847f8c949c48a4a8b087b8c0c8)
+            mstore(add(m, 0xa0), 0x16365272829aaec650acd0d28fdad4e22d6991bd97dfdcea58b4d6f29fede4f6)
+            mstore(add(m, 0xc0), 0xfe0f1f2f3f4b5b6b607b8b93a3a7b7bf357199c5abcfd9e168bcdee9b3f1ecf5)
+            mstore(add(m, 0xe0), 0xfd1e3e5a7a8aa2b670c4ced8bbe8f0f4fc3d79a1c3cde7effb78cce6facbf9f8)
+            mstore(0x40, add(m, 0x100))
+        }
+
+        for (uint256 i; i < 256; i++) {
+            uint256 x = 2**i;
+            assembly {
+                key := shr(248, mul(x, deBruijnSeq))
+                value := shr(248, mload(add(m, key)))
+            }
+            assertEq(value, i, "wrong value in lookup table");
+        }
+
+        uint256[] memory rows = DeBruijn.precomputeRows(deBruijnSeq);
+        uint256 rowFromLookupTable;
+        for (uint256 i; i < 8; i++) {
+            assembly {
+                rowFromLookupTable := mload(add(m, mul(32, i)))
+            }
+            assertEq(rowFromLookupTable, rows[i]);
+        }
+    }
```

Comment:
If I'm not mistaken, we're running a test against a test helper library (we cannot call these unit testing because these are not integrated with the true production code and thus cannot defend the role of unit testing the library, because the library's code can be desynced with the test code)

I'm not a fan of these, but we've already discussed it. Could we just move them to another test file, so that we separate test file concerns (and keep `TestMath.sol` only testing `Math.sol`?)

Reply:
It is still related to `Math.sol`, so it makes sense to me that it stays here.

But as you want, this is not important.

Reply:
The function was refactored in #88 to improve testing